From 7538ad166d0fb1999f32f2d80969184484c2ba07 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@digia.com>
Date: Mon, 27 Oct 2014 14:53:25 +0200
Subject: [PATCH] Update default args for apalis imx6

Boot from sd card by default, disable kernel logs from tty1, and
disable cursor blinking.
---
 include/configs/apalis_imx6.h | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/include/configs/apalis_imx6.h b/include/configs/apalis_imx6.h
index 5d32118..289d611 100644
--- a/include/configs/apalis_imx6.h
+++ b/include/configs/apalis_imx6.h
@@ -212,14 +212,14 @@
 		"&& setenv dtbparam \" - ${fdt_addr_r}\" && true\0"
 
 #define SD_BOOTCMD						\
-	"sdargs=ip=off root=/dev/mmcblk1p2 rw,noatime rootfstype=ext3 " \
-		"rootwait\0" \
+	"drive=2\0" \
 	"sdboot=" "run setup; " \
-		"setenv bootargs ${defargs} ${sdargs} ${setupargs} " \
+		"setenv bootargs ${defargs} ip=off root=/dev/mmcblk${drive}p2 " \
+		"rw,noatime rootfstype=ext3 rootwait ${setupargs} " \
 		"${vidargs}; echo Booting from SD card in 8bit slot...; " \
-		"run sddtbload; fatload mmc 1:1 ${kernel_addr_r} " \
+		"run sddtbload; fatload mmc ${drive}:1 ${kernel_addr_r} " \
 		"${boot_file} && bootm ${kernel_addr_r} ${dtbparam}\0" \
-	"sddtbload=setenv dtbparam; fatload mmc 1:1 ${fdt_addr_r} " \
+	"sddtbload=setenv dtbparam; fatload mmc ${drive}:1 ${fdt_addr_r} " \
 		"${fdt_file} && setenv dtbparam \" - ${fdt_addr_r}\" && true\0"
 
 #ifndef CONFIG_APALIS_IMX6_V1_0
@@ -229,7 +229,8 @@
 #define FDT_FILE "imx6q-apalis-eval_v1_0.dtb"
 #endif
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	"bootcmd=run emmcboot ; echo ; echo emmcboot failed ; " \
+	"bootcmd=run sdboot ; echo ; echo sdboot failed ; " \
+		"run emmcboot ; echo ; echo emmcboot failed ; " \
 		"run nfsboot ; echo ; echo nfsboot failed ; " \
 		"usb start ;" \
 		"setenv stdout serial,vga ; setenv stdin serial,usbkbd\0" \
@@ -242,7 +243,7 @@
 	NFS_BOOTCMD \
 	SD_BOOTCMD \
 	"setup=setenv setupargs fec_mac=${ethaddr} " \
-		"consoleblank=0  no_console_suspend=1 console=tty1 " \
+		"consoleblank=0  no_console_suspend=1 vt.global_cursor_default=0 " \
 		"console=${console},${baudrate}n8\0 " \
 	"setupdate=setenv drive 1; fatload mmc ${drive}:1 ${kernel_addr_r} " \
 		"flash_mmc.img || setenv drive 2; fatload mmc ${drive}:1 " \
-- 
1.9.1

